include/rocksdb/filter_policy.h

```diff
@@ -99,19 +100,32 @@ struct FilterBuildingContext {
   // Options for the table being built
   const BlockBasedTableOptions& table_options;
 
+  // BEGIN from (DB|ColumnFamily)Options in effect at table creation time
+  CompactionStyle compaction_style = kCompactionStyleLevel;
+
+  // Number of LSM levels, or -1 if unknown
+  int num_levels = -1;
+
+  // An optional logger for reporting errors, warnings, etc.
+  Logger* info_log = nullptr;
```

Comment:
Please forward declare Logger.  I know it is not pbly not needed based on the complicated includes, but it would be nice to one day clean those up...

Reply:
Not too concerned about this. Google style recommends against forward declarations, though we sometimes (maybe) need them to separate user-public from internal-only APIs. https://google.github.io/styleguide/cppguide.html#Forward_Declarations